backend/src/agents/datastore_agent.py

```diff
@@ -1,41 +1,52 @@
+from src.utils import call_model
+from src.utils.graph_db_utils import execute_query
+from src.agents import Agent, agent_metadata
+import logging
+from src.prompts import PromptEngine
+from datetime import datetime
+from src.utils import to_json
 from .tool import tool_metadata
 from .types import Parameter
-from src.agents import Agent, agent_metadata
 
+logger = logging.getLogger(__name__)
 
-@tool_metadata(
-    name="last month data",
-    description="Get spending for the last month on a merchant",
-    parameters={
-        "merchant_name": Parameter(
-            type="string",
-            description="Merchant name eg. Spotify",
-        )
-    },
-)
-def get_month_data(merchant_name: str) -> str:
-    if merchant_name.__contains__("Amazon"):
-        return "Last month you spend £64.21 on Amazon"
-    if merchant_name.__contains__("Netflix"):
-        return "Last month you spend £6.99 on Netflix"
-    return f"Last month you didn't spend anything on {merchant_name}"
+current_user = "John Doe"
+engine = PromptEngine()
+narrative_options = "Bills, Groceries, Entertainment, Rent, Shopping"
 
+graph_schema_prompt = engine.load_prompt("graph-schema", narrative_options=narrative_options)
 
-data_store_prompt = """
-You are an expert database agent. You can help with database queries and are connected to a neo4j database.
-
-When looking for an input to any tool being used, a single merchant should be provided.
-
-e.g. "Calculate the sum of Netflix spend over the last month"
-Merchant: Netflix
-"""
+generate_cypher_query_prompt = engine.load_prompt("generate-cypher-query",
+                                                  graph_schema_prompt=graph_schema_prompt,
+                                                  current_date=datetime.now())
 
+@tool_metadata(
+    name="generate cypher query",
+    description="Generate Cypher query based on user utterance",
+    parameters={"user_prompt": Parameter(
+            type="string",
+            description="The utterance from the user that the query will be based on",
+        )})
+
+def generate_query(user_prompt):
+        llm_query = call_model(generate_cypher_query_prompt, user_prompt)
+        logger.info("llm query: ")
+        logger.info(llm_query)
+        json_query = to_json(llm_query)
+        logger.info("Cypher generated by the LLM: ")
+        logger.info(json_query)
```

Comment:
do we need to log both here, is it repeated logs? Feel like the `json_query` is better to keep?

Reply:
yes good point Harry, since the llm is returning a json too, removed